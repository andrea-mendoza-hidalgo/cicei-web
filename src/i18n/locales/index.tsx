import en from './translations.en.json';
import es from './translations.es.json';
import { SharedEn, SharedEs } from '../../shared/components/locale';
import { PagesEn, PagesEs } from '../../components/pages/locales';
import ProjectEn from '../../components/project/locale/project.en.json';
import ProjectEs from '../../components/project/locale/project.es.json';
import PublicationsEn from '../../components/publication/locale/publication.en.json';
import PublicationsEs from '../../components/publication/locale/publication.es.json';
import ResearchersEn from '../../components/researchers/locale/researchers.en.json';
import ResearchersEs from '../../components/researchers/locale/researchers.es.json';
import NewsEs from '../../components/news/locale/news.es.json';
import NewsEn from '../../components/news/locale/news.en.json';

en.shared= SharedEn;
es.shared= SharedEs;
en.pages = PagesEn;
es.pages = PagesEs;
en.project = ProjectEn;
es.project = ProjectEs;
en.publication = PublicationsEn;
es.publication = PublicationsEs;
en.researchers = ResearchersEn;
es.researchers = ResearchersEs;
en.news = NewsEn;
es.news = NewsEs;



export {en, es};