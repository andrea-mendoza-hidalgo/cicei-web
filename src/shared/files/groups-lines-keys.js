export const GroupsKeys = {
    id: '_id',
    validator: 'groupValidator',
    name:'groupName'
}

export const LinesKeys = {
    id:'_id',
    validator:'lineValidator',
    name:'lineName'
}