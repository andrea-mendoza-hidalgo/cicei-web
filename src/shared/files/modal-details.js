export const ModalPublicationDetails = [
  { key: 'titlePublication', label: 'title', isLink: false},
  { key: 'autorPublication', label: 'author', isLink: false},
  { key: 'publishedIn', label: 'publishedIn', isLink: false},
  { key: 'yearPublication', label: 'yearPublication', isLink: false},
  { key: 'editorialPublication', label: 'editorial', isLink: false},
  { key: 'urlPublication', label: 'url', isLink: true},
  { key: 'abstractPublication', label: 'abstract', isLink: false}
];

export const ModalProjectDetails = [
  { key: 'projectCode', label: 'code' },
  { key: 'projectTitle', label: 'title' },
  { key: 'projectInvestigator', label: 'investigator' },
  { key: 'projectInstitution', label: 'institution' },
  { key: 'projectStudent', label: 'collaborator' },
  { key: 'projectDateStart', label: 'dateStart' },
  { key: 'projectAbstract', label: 'abstract' }
];
