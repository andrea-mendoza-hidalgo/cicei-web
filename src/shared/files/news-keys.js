export const NewsKeys = {
    id: '_id',
    owner: 'user_owner',
    title: 'title',
    date: 'date',
    resume: 'resume',
    name:'user_name',
    lastname: 'user_lastname'
};