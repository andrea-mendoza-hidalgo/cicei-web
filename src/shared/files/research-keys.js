export const ProjectsKeys = {
    id:'_id', 
    title: 'projectTitle',
    images: 'images',
    code: 'projectCode',
    researcher: 'projectInvestigator',
    student:'projectStudent',
    institution:'projectInstitution',
    abstract:'projectAbstract'
}

export const PublicationsKeys = {
    id:'_id',
    title:'titlePublication',
    author:'autorPublication',
    date:'yearPublication',
    editorial: 'editorialPublication'
}

export const ResearchersKeys = {
    id: '_id',
    name: 'user_name',
    lastname: 'user_lastname',
    degree: 'user_degree',
    group: 'user_group',
    mail: 'user_mail',
    line: 'user_line',
    resume: 'user_resume'
}