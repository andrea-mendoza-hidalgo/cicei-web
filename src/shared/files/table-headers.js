export const TablePublicationsHeaders = [
  { key: 'titlePublication', label: 'title' },
  { key: 'autorPublication', label: 'author' },
  { key: 'editorialPublication', label: 'editorial' },
  { key: 'yearPublication', label: 'yearPublication' }
];

export const TableProjectHeaders = [
  { key: 'projectCode', label: 'code' },
  { key: 'projectTitle', label: 'title' },
  { key: 'projectInvestigator', label: 'investigator' },
  { key: 'projectInstitution', label: 'institution' }
];

export const TableResearchersHeaders = [
  { key: 'user_lastname', label: 'lastname' },
  { key: 'user_group', label: 'group' },
  { key: 'user_line', label: 'lines' },
  { key: 'user_degree', label: 'degree' }
];