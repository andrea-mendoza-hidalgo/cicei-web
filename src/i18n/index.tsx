import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { es, en } from './locales';

const options = {
  debug: false,

  resources: {
    'es-ES': {
      shared: es.shared,
      pages: es.pages,
      project: es.project,
      publication: es.publication,
      researchers: es.researchers,
      news: es.news
    },
    'en-US': {
      shared: en.shared,
      pages: en.pages,
      project: en.project,
      publication: en.publication,
      researchers: en.researchers,
      news: en.news
    }
  },

  fallbackLng: 'es-ES',

  ns: ['shared', 'pages', 'project', 'publication'],

  defaultNS: 'shared',

  react: {
    wait: true
  }
};

i18n.use(initReactI18next).use(LanguageDetector).init(options);

export default i18n;
