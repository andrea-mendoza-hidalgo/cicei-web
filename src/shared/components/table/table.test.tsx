import React from 'react';
import { mount } from 'enzyme';
import { fireEvent, render, waitForElement, within } from '@testing-library/react';
import CustomTable from './table';
import { dataLink, list } from './table-data.mock';
import { TableProjectHeaders as headers, TableResearchersHeaders } from '../../files/table-headers';
import CustomLinkTable from './table-link';
import { MemoryRouter } from 'react-router-dom';
import '../../../i18n';

describe('Table Tests', () => {
  let table: any;
  const handleModalOpen = jest.fn();
  beforeEach(async () => {
    table = mount(<CustomTable headers={headers} data={list} handleClick={handleModalOpen} />);
  });

  afterEach(() => {
    table.unmount();
  });

  it('Table renders correctly ', () => {
    expect(table).toMatchSnapshot();
  });

  it('when screen size it is lower than 680 table has to show and open the collapse infomation', async () => {
    global.innerWidth = 680;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    const menu = table.find('.border__open');

    expect(menu.exists('.border__open-main-row')).toBeTruthy;
  });

  it('when screen size it is between 630 and 750 table has to show three columns', async () => {
    global.innerWidth = 700;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    expect(table).toMatchSnapshot();
  });

  it('when screen size it is between 420 and 629 table has to show three columns', async () => {
    global.innerWidth = 450;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    expect(table).toMatchSnapshot();
  });

  it('when screen size it is lower than 420, table has to show three columns', async () => {
    global.innerWidth = 320;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    expect(table).toMatchSnapshot();
  });

  it('when table is empty, mush show a empty table', async () => {
    const handleModalOpen = jest.fn();
    let wrapper = mount(<CustomTable headers={headers} data={[]} handleClick={handleModalOpen} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Table calls handleModalOpen method correctly ', () => {
    table.find('.table__body-cell').at(13).simulate('click');
    expect(handleModalOpen).toBeCalled();
  });

  it('Table make ascending sorting correctly', () => {
    table.find('.table__body-cell').at(13).simulate('click');
    expect(handleModalOpen).toBeCalled();
  });

  it('Table make descending sorting correctly', () => {
    table.find('.table__header-cell').at(3).simulate('click').simulate('click');
    expect(table).toMatchSnapshot();
  });
});

describe('Table with links Tests', () => {
  it('Table with link renders correctly ', () => {
    let table = mount(
      <MemoryRouter>
        <CustomLinkTable
          headers={TableResearchersHeaders}
          data={dataLink}
          linksColumn="user_lastname"
          loading= {false}
        />
      </MemoryRouter>
    );
    expect(table).toMatchSnapshot();
  });

  it('when screen size it is lower than 680 table has to show and open the collapse infomation in table link', async () => {
    let table = mount(
      <MemoryRouter>
        <CustomLinkTable
          headers={TableResearchersHeaders}
          data={dataLink}
          linksColumn="user_lastname"
          loading= {false}
        />
      </MemoryRouter>
    );

    global.innerWidth = 680;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    const menu = table.find('.border__open');

    expect(menu.exists('.border__open-main-row')).toBeTruthy;
  });

  it('when screen size it is between 630 and 750 table has to show three columns', async () => {
    let table = mount(
      <MemoryRouter>
        <CustomLinkTable
          headers={TableResearchersHeaders}
          data={dataLink}
          linksColumn="user_lastname"
          loading= {false}
        />
      </MemoryRouter>
    );
    global.innerWidth = 700;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    expect(table).toMatchSnapshot();
  });

  it('when screen size it is between 420 and 629 table has to show three columns', async () => {
    let table = mount(
      <MemoryRouter>
        <CustomLinkTable
          headers={TableResearchersHeaders}
          data={dataLink}
          linksColumn="user_lastname"
          loading= {false}
        />
      </MemoryRouter>
    );
    global.innerWidth = 450;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    expect(table).toMatchSnapshot();
  });

  it('when screen size it is lower than 420, table has to show three columns', async () => {
    let table = mount(
      <MemoryRouter>
        <CustomLinkTable
          headers={TableResearchersHeaders}
          data={dataLink}
          linksColumn="user_lastname"
          loading= {false}
        />
      </MemoryRouter>
    );
    global.innerWidth = 320;
    global.dispatchEvent(new Event('resize'));

    table.find('.table__mobile').at(3).simulate('click');

    expect(table).toMatchSnapshot();
  });

  it('Table make descending sorting correctly', () => {
    let table = mount(
      <MemoryRouter>
        <CustomLinkTable
          headers={TableResearchersHeaders}
          data={dataLink}
          linksColumn="user_lastname"
          loading= {false}
        />
      </MemoryRouter>
    );
    table.find('.table__header-cell').at(3).simulate('click').simulate('click');
    expect(table).toMatchSnapshot();
  });
});
