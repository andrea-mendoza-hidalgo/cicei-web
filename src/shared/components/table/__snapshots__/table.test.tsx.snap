// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Tests Table make descending sorting correctly 1`] = `ReactWrapper {}`;

exports[`Table Tests Table renders correctly  1`] = `ReactWrapper {}`;

exports[`Table Tests when screen size it is between 420 and 629 table has to show three columns 1`] = `ReactWrapper {}`;

exports[`Table Tests when screen size it is between 630 and 750 table has to show three columns 1`] = `ReactWrapper {}`;

exports[`Table Tests when screen size it is lower than 420, table has to show three columns 1`] = `ReactWrapper {}`;

exports[`Table Tests when table is empty, mush show a empty table 1`] = `ReactWrapper {}`;

exports[`Table with links Tests Table make descending sorting correctly 1`] = `ReactWrapper {}`;

exports[`Table with links Tests Table with link renders correctly  1`] = `ReactWrapper {}`;

exports[`Table with links Tests when screen size it is between 420 and 629 table has to show three columns 1`] = `ReactWrapper {}`;

exports[`Table with links Tests when screen size it is between 630 and 750 table has to show three columns 1`] = `ReactWrapper {}`;

exports[`Table with links Tests when screen size it is lower than 420, table has to show three columns 1`] = `ReactWrapper {}`;
